lua/customroles/admin.lua

```diff
@@ -0,0 +1,355 @@
+        draw.SimpleText(LANG.GetTranslation("admin_power_title"), "HealthAmmo", 30, ScrH() - 59, Color(0, 0, 10, 200), TEXT_ALIGN_LEFT)
+        CRHUD:ShadowedText(tostring(current_power), "HealthAmmo", 230, ScrH() - 59, COLOR_WHITE, TEXT_ALIGN_RIGHT)
+    end)
+
+    --------------
+    -- TUTORIAL --
+    --------------
+
+    hook.Add("TTTTutorialRoleText", "Admin_TTTTutorialRoleText", function(role, titleLabel)
+        if role == ROLE_ADMIN then
+            local roleColor = ROLE_COLORS[ROLE_ADMIN]
+            local detectiveColor = ROLE_COLORS[ROLE_DETECTIVE]
+
+            local html = "The " .. ROLE_STRINGS[ROLE_ADMIN] .. " is a " .. ROLE_STRINGS[ROLE_DETECTIVE] .. " and a member of the <span style='color: rgb(" .. roleColor.r .. ", " .. roleColor.g .. ", " .. roleColor.b .. ")'>innocent team</span> whose job is to find and eliminate their enemies."
+
+            html = html .. "<span style='display: block; margin-top: 10px;'>Instead of getting a DNA Scanner like a vanilla <span style='color: rgb(" .. detectiveColor.r .. ", " .. detectiveColor.g .. ", " .. detectiveColor.b .. ")'>" .. ROLE_STRINGS[ROLE_DETECTIVE] .. "</span>, they have the ability to open an admin menu and use powerful commands. Commands require admin power which generates over time.</span>"
+
+            return html
+        end
+    end)
+
+    -------------------
+    -- BLIND OVERLAY --
+    -------------------
+
+    net.Receive("TTT_AdminBlindClient", function()
+        if net.ReadBool() then
+            hook.Add("HUDPaint", "Admin_HUDPaint_Blind", function()
+                draw.RoundedBox(0, 0, 0, ScrW(), ScrH(), Color(0, 0, 0, 255))
+            end)
+        else
+            hook.Remove("HUDPaint", "Admin_HUDPaint_Blind")
+        end
+    end)
+
+    ----------------------
+    -- FAKE KICK SCREEN --
+    ----------------------
+
+    surface.CreateFont("KickText", {
+        font = "Tahoma",
+        size = 18,
+        weight = 400,
+        antialias = false
+    })
+
+    net.Receive("TTT_AdminKickClient", function()
+        local client = LocalPlayer()
+
+        hook.Add("HUDShouldDraw", "Admin_HUDShouldDraw_Kick", function(name)
+            if name ~= "CHudGMod" then return false end
+        end)
+
+        hook.Add("PlayerBindPress", "Admin_PlayerBindPress_Kick", function(ply, bind, pressed)
+            if (string.find(bind, "+showscores")) then return true end
+        end)
+
+        hook.Add("Think", "Admin_Think_Kick", function()
+            client:ConCommand("soundfade 100 1")
+        end)
+
+
+        local dframe = vgui.Create("DFrame")
+        dframe:SetSize(ScrW(), ScrH())
+        dframe:SetPos(0, 0)
+        dframe:SetTitle("")
+        dframe:SetVisible(true)
+        dframe:SetDraggable(false)
+        dframe:ShowCloseButton(false)
+        dframe:SetMouseInputEnabled(true)
+        dframe:SetDeleteOnClose(true)
+        dframe.Paint = function() end
+
+        local mat = Material("ui/roles/adm/kickScreen.png")
```

Comment:
Wiki says this call can be expensive and it should be cached. Though I guess this isn't being called repeated so it's probably fine?

Reply:
Yeah its probably not a big issue since it will be so rarely called, but its easy enough to cache so might as well